dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/dataentry/DataSetValueController.java

```diff
@@ -0,0 +1,93 @@
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.webapi.controller.dataentry;
+
+import static org.hisp.dhis.commons.collection.CollectionUtils.mapToList;
+
+import java.util.List;
+import java.util.Set;
+
+import lombok.RequiredArgsConstructor;
+
+import org.hisp.dhis.category.CategoryOptionCombo;
+import org.hisp.dhis.common.DhisApiVersion;
+import org.hisp.dhis.dataset.DataSet;
+import org.hisp.dhis.datavalue.DataExportParams;
+import org.hisp.dhis.datavalue.DataValue;
+import org.hisp.dhis.datavalue.DataValueService;
+import org.hisp.dhis.minmax.MinMaxDataElement;
+import org.hisp.dhis.minmax.MinMaxDataElementService;
+import org.hisp.dhis.organisationunit.OrganisationUnit;
+import org.hisp.dhis.period.Period;
+import org.hisp.dhis.webapi.controller.datavalue.DataValidator;
+import org.hisp.dhis.webapi.mvc.annotation.ApiVersion;
+import org.hisp.dhis.webapi.webdomain.datavalue.DataSetValueQueryParams;
+import org.hisp.dhis.webapi.webdomain.datavalue.DataValueDtoMapper;
+import org.hisp.dhis.webapi.webdomain.datavalue.DataValuesDto;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+/**
+ * @author Lars Helge Overland
+ */
+@Controller
+@RequiredArgsConstructor
+@RequestMapping( "/dataEntry" )
+@ApiVersion( { DhisApiVersion.DEFAULT, DhisApiVersion.ALL } )
+public class DataSetValueController
+{
+    private final DataValueService dataValueService;
+
+    private final MinMaxDataElementService minMaxValueService;
+
+    private final DataValidator dataValidator;
+
+    @GetMapping( "/dataValues" )
+    public DataValuesDto getDataValueSet( DataSetValueQueryParams params )
+    {
+        DataSet ds = dataValidator.getAndValidateDataSet( params.getDs() );
+        Period pe = dataValidator.getAndValidatePeriod( params.getPe() );
+        OrganisationUnit ou = dataValidator.getAndValidateOrganisationUnit( params.getOu() );
+        CategoryOptionCombo ao = dataValidator.getAndValidateAttributeOptionCombo( params.getCc(), params.getCp() );
```

Comment:
Makes me think a validation for `DataSetValueQueryParams` would be nicer but I guess this is also used as a way to map uids to objects.